### ISSUE TITLE
Various static analysis warnings (by  @Syphurith)

### ISSUE THREAD

**Issue by Bigpet** [RPCS3/rpcs3#1064]
Alright, since we have too many `TODO:`s strewn about in the code I want to highlight these to stay visible after #1059 is merged
- [ ] [always false if (unsigned < 0)](https://github.com/RPCS3/rpcs3/blob/fd34d11eec5069f8d39b51bab67c316ae6c057eb/rpcs3/Crypto/lz.cpp#L143)
- [ ] [unsigned <=0 , maybe it should be ==0 maybe something else](https://github.com/RPCS3/rpcs3/blob/fd34d11eec5069f8d39b51bab67c316ae6c057eb/rpcs3/Crypto/sha1.cpp#L236)
- [ ] [= and == precedence is probably wrong, some braces would help](https://github.com/RPCS3/rpcs3/blob/fd34d11eec5069f8d39b51bab67c316ae6c057eb/rpcs3/Emu/ARMv7/Modules/sceLibc.cpp#L31)
- [ ] [std::remove_if without erase](https://github.com/RPCS3/rpcs3/blob/fd34d11eec5069f8d39b51bab67c316ae6c057eb/rpcs3/Emu/Cell/PPULLVMRecompiler.cpp#L5823)
- [ ] [if and else if with identical condition](https://github.com/RPCS3/rpcs3/blob/fd34d11eec5069f8d39b51bab67c316ae6c057eb/rpcs3/Emu/SysCalls/Modules/cellAdec.cpp#L199)
- [ ] [another if and else if with same condition](https://github.com/RPCS3/rpcs3/blob/fd34d11eec5069f8d39b51bab67c316ae6c057eb/rpcs3/Emu/SysCalls/Modules/cellVdec.cpp#L171)
- [ ] [check if discarded return values are important](https://github.com/RPCS3/rpcs3/blob/fd34d11eec5069f8d39b51bab67c316ae6c057eb/rpcs3/Gui/InterpreterDisAsm.cpp#L517-L522)

**Comment by Syphurith** [comment]
@Bigpet Well if anyone would like to run valgrind memchecks there may be more to be fixed.

**Comment by Syphurith** [comment]
Valgrind ran.. [File](https://github.com/RPCS3/rpcs3/issues/1070#issuecomment-97875232). Have read the huge log (many of which are from glibc, fcitx, gtk..) entirely and find out rpcs3 is written well. Only one definitely lost in `rpcs3/Gui/GameView.h`, that is `m_img_list` which isn't released - but it is in the main window so i could consider it as not harmful..
I can not get those `gcm_*` to run cause 3D accel enable failed. Since valgrind is a runtime checking tool, maybe someone else could run it further. BTW what about Visual Leak Detector (vld)?

**Comment by paulsapps** [comment]
One idea for the future is add null rendering backend so on travis you can run valgrind against hello world app for say 5 seconds to check main code paths dont have any obvious memory bugs

**Comment by necros2k7** [comment]
[rpcs3\Emu\Io\MouseHandler.h:122]: (performance) Variable 'x_pos' is assigned in constructor body. Consider performing initialization in initialization list.
[rpcs3\ds4_pad_handler.cpp:154]: (performance) Function parameter 'buttons' should be passed by const reference.
[rpcs3\Emu\RSX\Common\texture_cache_utils.h:188]: (performance) Prefer prefix ++/-- operators for non-primitive types.
[rpcs3\rpcs3qt\register_editor_dialog.cpp:104]: (performance) Inefficient usage of string::find() in condition; string::compare() would be faster.

**Comment by Bigpet** [comment]
@necros2k7  please open a new issue this one's outdated and I'll close it
